### PR TITLE
Add ruby parser implementation

### PR DESCRIPTION
**Pull request by greggroth**

### CODE REVIEW

File: ruby/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+The ruby gem currently supports loading and applying a YAML decision tree.  The
+following example is from the tests and shows how a YAML fixture file is parsed
+into a `BinaryTree` object:
+
+```ruby
+tree = TreeThinking::BinaryTreeFactory.from_yaml('./spec/fixtures/simple_tree.yaml')
+```
+
+This tree can then be used to get a probability from an answer vectory:
```
Comment (BBischof) [inline ruby/README.md]: Typo!

File: ruby/lib/tree_thinking/binary_tree_factory.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      btree.tree = nodify(tree['tree'])
+      btree
+    end
+
+    private
+
+    def nodify(attrs)
+      BinaryTree::Node.new(
+        attrs['feature_idx'],
+        attrs['op'],
```
Comment (greggroth) [inline ruby/lib/tree_thinking/binary_tree_factory.rb]: @BBischof I didn't include a default operator, so this is counting on the operator being specified explicitly.

Comment (BBischof) [inline ruby/lib/tree_thinking/binary_tree_factory.rb]: Would that be difficult to add? I think the default is something we definitely want to do.

Comment (greggroth) [inline ruby/lib/tree_thinking/binary_tree_factory.rb]: It's not difficult, but it sounds like there isn't a reasonable default to assume, right?

Comment (BBischof) [inline ruby/lib/tree_thinking/binary_tree_factory.rb]: I think `<=` should be the default. Specifically because it's the convention for DTs.

Comment (greggroth) [inline ruby/lib/tree_thinking/binary_tree_factory.rb]: _done_  6688620

Comment (BBischof) [inline ruby/lib/tree_thinking/binary_tree_factory.rb]: 👍

**Pull request merged**